### ISSUE TITLE
old setuptools versions incurr broken version strings with bad warnings

### ISSUE THREAD

**Issue by yun** [pypa/setuptools-scm#169]
A package version containing a revision hash seems to break  `pip` when using `--pre` flag.
This is easy to demonstrate by using `pkg_resources.parse_version`:
```python
In [1]: from pkg_resources import parse_version

# with commit hash in version, does not detect it as a prerelease.
In [2]: parse_version('0.1.2.dev20-gef3607f').is_prerelease
Out[2]: False

# without commit hash it works as intended, prerelease = True.
In [3]: parse_version('0.1.2.dev20').is_prerelease
Out[3]: True
```

See also PEP440 here: https://www.python.org/dev/peps/pep-0440/#dvcs-based-version-labels

I think the proper fix is to separate the revision hash with a '+' character to make it part of the local  version instead of public:
```python
In [1]: from pkg_resources import parse_version

In [2]: a = parse_version('0.1.2.dev20+gef3607f')

In [3]: a.is_prerelease
Out[3]: True

In [4]: a.public
Out[4]: '0.1.2.dev20'

In [5]: a.local
Out[5]: 'gef3607f'
```

**Comment by yun** [comment]
Nevermind, it seems my setuptools was outdated and thus did not use '+' as seperator.

**Comment by RonnyPfannschmidt** [comment]
@yun what version of setuptools where you using, depending on its age i may still need to act

**Comment by yun** [comment]
Version was 0.9.8, which is the default CentOS 7 package of python-setuptools.

python-setuptools-0.9.8-4.el7.noarch

I upgraded to 34.3.2 using pip.

**Comment by RonnyPfannschmidt** [comment]
@yun was there a warning being printed? (i recall it should, but if it has gone missing i need to investigate and repair)

**Comment by yun** [comment]
The only warning was:

```
your setuptools is too old (<12)
setuptools_scm is degraded
```

**Comment by RonnyPfannschmidt** [comment]
@yun thanks for the update, do you as a user being at the receiving end of that message thing it is ok or should it be improved?

**Comment by yun** [comment]
The error message was kinda lost in the rest of the output of `setup.py sdist`. I think it could be improved by documenting it on the main page, what 'degraded' mode means.

For example, I did not expect that it would cause it to generate a different version string because of it.

**Comment by RonnyPfannschmidt** [comment]
@jaraco i do wonder if this is something that should be dealt with discretely or if a bigger error is acceptable there - any oppinion?

**Comment by RonnyPfannschmidt** [comment]
ref #241

**Comment by RonnyPfannschmidt** [comment]
fixed by #241
